crates/bevy_ecs/src/system/query.rs

```diff
@@ -975,6 +975,44 @@ where
         self.state
             .is_empty(self.world, self.last_change_tick, self.change_tick)
     }
+
+    /// Returns `true` if the given [`Entity`] matches the query.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct Player;
+    /// # #[derive(Component)]
+    /// # struct Alive;
+    /// fn check_if_alive(
+    ///     player_query: Query<Entity, With<Player>>,
+    ///     alive_query: Query<(), With<Alive>>,
+    /// ) {
+    ///     if let Ok(player) = player_query.get_single() {
```

Comment:
Hmm, so I think a more idiomatic version of this would use https://github.com/bevyengine/bevy/issues/3088 (or the option workaroud detailed therein). I think this method is more useful for pseudo-relations, e.g. 'has this pet's owning player been killed', or ['how many neighbours of this entity are alive'](https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life).

However, I'm struggling to think of a super succinct version which does this.

Reply:
Also, replacing the get_single with `single` would be a welcome change.

Reply:
Here's an alternate example to steal that's a bit more idiomatic. 

```rust
#[derive(Component)]
struct InRange;

struct Target {
    entity: Entity,
}

fn targeting_system(in_range_query: Query<&InRange>, target: Res<Target>) {
    if in_range_query.contains(target.entity) {
        println!("Bam!")
    }
}
```